feat(register): always resolve full file path for tsconfig and add paths if available

dmregister opened the pull request:
Addresses #626

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/swc-project/swc-node?pullRequest=646) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/swc-project/swc-node?pullRequest=646) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/swc-project/swc-node?pullRequest=646) it.</sub>

dmregister commented:
@Brooooooklyn I'm not sure how these snapshots work. I added the snapshot file on the last commit and it failed as it was not used. Once I deleted it, now it's failing cause it's missing. Any idea how to resolve this? I do not get the error locally.

dmregister commented:
@Brooooooklyn let me know if this PR needs anything else or if I need to split this up into 2 PR's
1. for the paths updates
2. fully resolve tsConfig path
